src/js/components/sharedComponents/VerticalLine.jsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * VerticalTodayLine.jsx
+ * Created by Jonathan Hill 07/05/19
+ **/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { throttle } from 'lodash';
+
+const propTypes = {
+    xScale: PropTypes.func, // function to set line position
+    text: PropTypes.string, // text to display with line
+    y1: PropTypes.number, // top of graph
+    y2: PropTypes.number, // bottom of graph
+    xMax: PropTypes.number, // max possible value of x
+    xMin: PropTypes.number, // max possible value of x
+    xValue: PropTypes.number, // actual value of x
+    showTextRight: PropTypes.bool, // show text to the right of line
+    showTextLeft: PropTypes.bool, // show text to the left of line
+    textY: PropTypes.number, // show text at this height
+    description: PropTypes.string
+};
+
+export default class VerticalLine extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            textX: 0,
+            textY: props.textY,
+            windowWidth: null
+        };
+        this.handleWindowResize = throttle(this.handleWindowResize.bind(this), 50);
+        this.textDiv = null;
+        // React will call this function when the DOM draws it ( React Callback Refs )
+        this.setTextDiv = (element) => {
```

Comment:
This is great -- is there anywhere else we should be doing this? I recommend adding TODO's to that effect as I think there certainly are other places where we aren't doing this!

Reply:
@MaxwellKendall I do not know, I think distinguishing what needs to be changed in the app should be done if time is allowed at the end of this sprint.